gems/pending/spec/appliance_console/database_replication_spec.rb

```diff
@@ -0,0 +1,167 @@
+
+      expect(subject).to receive(:just_ask).with(/ username/i, 'defaultuser').and_return("newuser")
+      expect(subject).to receive(:just_ask).with(/ name/i, 'defaultdatabasename').and_return("newuser")
+      expect(subject)
+        .to receive(:ask_for_ip_or_hostname).with(/primary.*hostname/i, 'defaultprimary').and_return("newprimary")
+      expect(subject).to receive(:ask_for_password_or_none).with(/password/i, anything).and_return("the password")
+      expect(subject).to receive(:ask_for_password).with(/password/i).and_return("the password")
+
+      subject.ask_for_database_credentials
+    end
+  end
+
+  context "#confirm" do
+    it "should ask for standby host when requested" do
+      subject.standby_host = 'defaultstandby'
+
+      expect(subject).to receive(:clear_screen)
+      expect(subject).to receive(:say).with(/^replication/i)
+      expect(subject).to receive(:say).with(/standby.*defaultstandby/i)
+      expect(subject).to receive(:agree).with(/^apply/i)
+      subject.confirm(:including_standby_host)
+    end
+
+    it "should not ask for standby host if not requested" do
+      expect(subject).to receive(:clear_screen)
+      expect(subject).to receive(:say).with(/^replication/i)
+      expect(subject).to_not receive(:say).with(/standby.*defaultstandby/i)
+      expect(subject).to receive(:agree).with(/^apply/i)
+      subject.confirm
+    end
+  end
+
+  context "#confirm_reconfiguration" do
+    it "should log a warning and ask to continue anyway" do
+      expect(subject).to receive(:say).with(/^warning/i)
+      expected_logging = double
+      subject.logger = expected_logging
+      expect(expected_logging).to receive(:info).with(/^warning/i)
+      expect(subject).to receive(:agree).with(/^continue/i)
+
+      subject.confirm_reconfiguration
+    end
+  end
+
+  context "#create_config_file" do
+    before do
+      subject.cluster_name      = 'clustername'
+      subject.node_number       = 'nodenumber'
+      subject.database_name     = 'databasename'
+      subject.database_user     = 'user'
+
+      @temp_file = Tempfile.new(subject.class.name.split('::').last.downcase)
+      stub_const("ApplianceConsole::DatabaseReplication::REPMGR_CONFIG", @temp_file.path)
+    end
+
+    after do
+      @temp_file.close
+      @temp_file.unlink
+    end
+
+    it "should set the cluster parameter in the config file" do
+      subject.create_config_file('host')
+      expect(File.read(@temp_file.path)).to include("cluster=clustername")
+    end
+
+    it "should set the node parameter in the config file" do
+      subject.create_config_file('host')
+      expect(File.read(@temp_file.path)).to include("node=nodenumber")
+    end
+
+    it "should set the node_name parameter in the config file" do
+      subject.create_config_file('host')
+      expect(File.read(@temp_file.path)).to include("node_name=host")
+    end
+
+    it "should set the conninfo parameter in the config file" do
```

Comment:
You can combine this test and the above three and expect the contents of the file exactly. That way we're also testing that `use_replication_slots` and `pg_basebackup_options` are also set.